titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeUtil.java

```diff
@@ -148,4 +161,48 @@ public static String formatV1ContainerState(V1ContainerState containerState) {
 
         return "{state=<not set>}";
     }
+
+    /**
+     * If a job has an availability zone hard constraint with a farzone id, return this farzone id.
+     */
+    public static Optional<String> findFarzoneId(DirectKubeConfiguration configuration, Job job) {
+        List<String> farzones = configuration.getFarzones();
```

Comment:
could this be generalized to any availabilityZones? Does it matter that it is a farzone? I'd recommend naming this `isAvailabilityZoneOwnedByKubeScheduler()` and `configuration.getAvailabilityZonesOwnedByKubeScheduler()`, or similar.

IOW, can we abstract away that some AZs are farzones?

Reply:
It is more than an additional zone. Farzones are excluded from the default scheduling, and there is extra logic around that. I would prefer to keep it explicit.